server/index.js

```diff
@@ -55,7 +55,9 @@ export async function createServer(
       console.log(`Webhook processed, returned status code 200`);
     } catch (error) {
       console.log(`Failed to process webhook: ${error}`);
-      res.status(500).send(error.message);
+      if (!res.headersSent) {
+        res.status(500).send(error.message);
```

Comment:
If a header was sent at some point, but we still got an error, shouldn't we still send a status code back? I wonder if this could end up in a situation where the response is never sent.

Reply:
Hmmm if headers have been sent, then hasn't the status code been sent as well already? Based on [Express docs](https://expressjs.com/en/api.html#res.headersSent), `headersSent` wouldn't be true until a response was sent. I can pull together some quick tests though when I have a minute.

The main issue here is that the `process` method of `WebhooksRegistry` has a few instances where it [completes the response](https://github.com/Shopify/shopify-node-api/blob/main/src/webhooks/registry.ts#L437-L438).

Reply:
I think `process` will have completed the response in all instances ... there's a
```
          response.writeHead(<statusCode>);
          response.end();
```
before every `resolve` or `reject` of the return `Promise` in `process`.

So I think the solution would be to actually remove the 
```
res.status(500).send(error.message);
```
line entirely.

Reply:
☝🏻 If removing that line is a bit too scary, then checking the `res.headersSent` is a good sanity check.